x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/IndexDeprecationChecks.java

```diff
@@ -574,4 +576,33 @@ static DeprecationIssue emptyDataTierPreferenceCheck(ClusterState clusterState,
         }
         return null;
     }
+
+    static DeprecationIssue checkSettingNoReplacement(IndexMetadata indexMetadata, Setting<?> deprecatedSetting, String url) {
+        assert deprecatedSetting.isDeprecated() : deprecatedSetting;
+        if (deprecatedSetting.exists(indexMetadata.getSettings()) == false) {
+            return null;
+        }
+        final String deprecatedSettingKey = deprecatedSetting.getKey();
+        final String message = String.format(Locale.ROOT, "Setting [%s] is deprecated", deprecatedSettingKey);
+        final String details = String.format(Locale.ROOT, "Remove the [%s] setting.", deprecatedSetting.getKey());
+        return new DeprecationIssue(DeprecationIssue.Level.CRITICAL, message, url, details, false, null);
```

Comment:
I think this could delegate to the other `checkSettingNoReplacement` that is in `NodeDeprecationChecks` by calling:

```java
return checkSettingNoReplacement(indexMetadata.getSettings(), deprecatedSetting, url)
```

So that it doesn't duplicate the body, since they are the same.